### PR TITLE
ImGuizmo sandbox visual scripting nodes

### PR DESCRIPTION
**Pull request by LiamANeeson**
- Updated Node Editor Using ImGuizmo and ImGui.
- Gives ability to Customise Nodes and the Editor Screen
- Dynamically Draw Links between Nodes.
- Basis for Editor Layer Visual Scripting

### CODE REVIEW

File: HorizonEngine/src/HorizonEngine/GraphEditor/GraphEditor.h [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        float mBorderThickness{ 6.f }; // thickness of selection border around nodes
+        float mNodeSlotRadius{ 8.f }; // circle radius for inputs and outputs
+        float mNodeSlotHoverFactor{ 1.2f }; // increase size when hovering
+        float mMinZoom{ 0.2f }, mMaxZoom{ 1.1f };
+        float mSnap{ 5.f };
+        bool mDisplayLinksAsCurves{ true }; // false is straight and 45deg lines
+        bool mAllowQuadSelection{ true }; // multiple selection using drag and drop
+        bool mRenderGrid{ true }; // grid or nothing
+        bool mDrawIONameOnHover{ true }; // only draw node input/output when hovering
+    };
```
Comment (alex-mulkerrins) [inline HorizonEngine/src/HorizonEngine/GraphEditor/GraphEditor.h:51]: I like the comments here, really helpful to understand what each line does

File: HorizonEngine/src/HorizonEngine/GraphEditor/GraphEditor.h [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    typedef size_t TemplateIndex;
+
+    // Force the view to be respositionned and zoom to fit nodes with Show function.
+    // Parameter value will be changed to Fit_None by the function.
+    enum FitOnScreen
+    {
+        Fit_None,
+        Fit_AllNodes,
+        Fit_SelectedNodes
+    };
```
Comment (alex-mulkerrins) [inline HorizonEngine/src/HorizonEngine/GraphEditor/GraphEditor.h:21]: Ah this is cool, so it will adjust to show all the nodes?

Comment (LiamANeeson) [reply HorizonEngine/src/HorizonEngine/GraphEditor/GraphEditor.h:21]: Yeah, exactly can fit highlight nodes and then all nodes

Comment (alex-mulkerrins) [reply HorizonEngine/src/HorizonEngine/GraphEditor/GraphEditor.h:21]: sweet, look forward to having a mess around with it myself!

**Review by TejasJD — APPROVED** [review]

**Review by Minghui-Gao — APPROVED** [review]

**Review by BozhidarVelikov — APPROVED** [review]

**Review by alex-mulkerrins — APPROVED** [review]

**Pull request merged**